### PR TITLE
Support for mock as function of incoming request body

### PR DESCRIPTION
**Pull request by Maczaj**
Hello,
although the tool does great job, lack of ability to conditionally build response based on incoming request is a big drawback in my current project (I have multiple scenarios that can happen depending on server's response and I'd like to have a way to somehow trigger all of them)

### CODE REVIEW

**Comment by odino** [comment]
Hey @Maczaj  thanks for this! @mohdhallal could you have a look? :)

**Comment by mohdhallal** [comment]
Hello guys. @odino thanks for the tag.

@Maczaj thanks a lot for the effort.

Just to get on the same page, can you help me with a couple of use-cases to understand when you would need such flexibility?

The way I see it, if you want the server to return different responses, you can use different headers, different query params, different POST params, etc... why aren't the existing features flexible enough for your project?

On the other hand, this seems a valid feature to extend the `#import` syntax. Taking a quick look at the codebase (haven't touched it in a while). I think we can pass a set of standard arguments (including the request body) in https://github.com/namshi/mockserver/blob/master/mockserver.js#L88 then you (and others) would get the ultimate flexibility to vary responses based on anything passed in the original request.

wdyt?

**Comment by Maczaj** [comment]
First of all thanks for picking this so quick :)

@mohdhallal Sure, I'll try to explain my use case. Generally, I'm using mockserver for our QA engineer working on test automation with Selenium. He needs a way to go through all screens to prepare test scripts before executing them against real systems. Selenium only interacts with UI, thus technical stuff such as adding custom headers is not possible. Different post params also won't do the trick since we're using Windows (corpo rules :( ). With solution from this PR it will be possible for us to trigger all possible scenarios.

Regarding your suggestion - I think it's good idea, I'll work on it tomorrow

File: README.md [hunk 1]
```diff
@@ -258,20 +258,21 @@ module.exports = {
...
+X-Subject-Token: #header ${require('uuid/v4')()};
+```
 You can also use function to build dynamic response:
 ```js
 // script.js
-module.exports = function(incomingBody) {
+module.exports = function(request) {
     return {
-        prop: incomingBody.foo ? 'bar' : 'baz'
+        prop: request.body.indexOf('foo') !== -1 ? 'bar' : 'baz'
```
Comment (cirpo) [inline README.md]: what about `request.body.includes('foo') ? 'bar' : 'baz'`

File: mockserver.js [hunk 2]
```diff
@@ -52,7 +52,7 @@ var prepareWatchedHeaders = function () {
  * returning an HTTP-ish object with
  * status code, headers and body.
  */
-var parse = function (content, file, originalBody) {
+var parse = function (content, file, requestData) {
```
Comment (cirpo) [inline mockserver.js]: I would keep `request` instead of `requestData`, data is implicit imo

**Comment by mohdhallal** [comment]
@Maczaj @odino @cirpo take a look here please: https://github.com/namshi/mockserver/pull/57

we would have access to the `request` param in the js file by simply passing it to the `parse` function. This has to do with `eval` executed within the local scope of `parse`

**Comment by Maczaj** [comment]
@mohdhallal Your version is better, I guess we can close this one :)

**Pull request closed without merging**